paasta_tools/cli/cmds/validate.py

```diff
@@ -728,30 +802,18 @@ def paasta_validate_soa_configs(
     if not validate_service_name(service):
         return False
 
-    returncode = True
-
-    if not validate_all_schemas(service_path):
-        returncode = False
-
-    if not validate_tron(service_path, verbose):
-        returncode = False
-
-    if not validate_paasta_objects(service_path):
-        returncode = False
-
-    if not validate_unique_instance_names(service_path):
-        returncode = False
-
-    if not validate_autoscaling_configs(service_path):
-        returncode = False
-
-    if not validate_secrets(service_path):
-        returncode = False
-
-    if not validate_min_max_instances(service_path):
-        returncode = False
-
-    return returncode
+    checks = [
+        validate_all_schemas,
+        partial(validate_tron, verbose=verbose),
+        validate_paasta_objects,
+        validate_unique_instance_names,
+        validate_autoscaling_configs,
+        validate_secrets,
+        validate_min_max_instances,
+        validate_cpu_burst,
+    ]
+
+    return all(check(service_path) for check in checks)
```

Comment:
this is mostly me being cheeky - there's a bit of a behavior change here in that now we'll stop running checks as soon as one fails, but in a way that seems fine to me? e.g., if your files don't pass schema validation: we can't really assume that the rest of the checks will even have valid config files to look at

that said: i can undo this :p

Reply:
I think it's fine to abort later checks if a failure in an early check would probably cause later checks to fail too, and the checks are sorted by priority/informativeness.

`validate_tron` seems pretty unrelated to `validate_paasta_objects` though.

You could restore the original, non-short-circuiting behavior by doing `all([...])` instead of `all(...)`.

Reply:
(ended up restoring the original behavior so that i didn't need to think about the optimal order here)